src/mlpack/core/optimizers/cmaes/cmaes.hpp

```diff
@@ -0,0 +1,347 @@
+/**
+ * @file cmaes.hpp
+ * @author Kartik Nighania (GSoC 17 mentor Marcus Edel)
+ *
+ * Covariance Matrix Adaptation Evolution Strategy
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_CMAES_CMAES_HPP
+#define MLPACK_CORE_OPTIMIZERS_CMAES_CMAES_HPP
+
+#include <mlpack/core.hpp>
+#include <cassert>
```

Comment:
Most of the following header files are already included or not needed anymore can you go through the list and remove the unnecessary includes?